pkg/flow/logging/level/level.go

```diff
@@ -0,0 +1,50 @@
+package level
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/flow/logging"
+)
+
+const (
+	levelKey = "level"
+)
+
+// Error returns a logger that includes a Key/ErrorValue pair.
+func Error(logger log.Logger) log.Logger {
+	return toLevel(logger, "error", slog.LevelError)
+}
+
+// Warn returns a logger that includes a Key/WarnValue pair.
+func Warn(logger log.Logger) log.Logger {
+	return toLevel(logger, "warn", slog.LevelWarn)
+}
+
+// Info returns a logger that includes a Key/InfoValue pair.
+func Info(logger log.Logger) log.Logger {
+	return toLevel(logger, "info", slog.LevelInfo)
+}
+
+// Debug returns a logger that includes a Key/DebugValue pair.
+func Debug(logger log.Logger) log.Logger {
+	return toLevel(logger, "debug", slog.LevelDebug)
+}
+
+func toLevel(logger log.Logger, level string, slogLevel slog.Level) log.Logger {
+	if adapter, ok := logger.(logging.EnabledAware); ok {
```

Comment:
IIRC each time an interface gets resolved it needs an allocation (maybe that's why even the `Logging_Slog_Drops` case allocates).

I'm curious, what does a `switch logger.(type)` performance look like? Is that the same here?

Reply:
I have switched to the implementation you've suggested, but there are no improvements in performance. I will leave it, however, since I like the switch statement more :) 

Re: `Logging_Slog_Drops` I did manage to reduce one allocation in the benchmark code due to `time.Since(now)`, so now it does 0 allocations.

Reply:
Hm, interesting tidbit to keep in mind!

I wouldn't expect that `x.(type)` would be the same performance if it didn't have to look up an interface value. No personal preference on the stylistic front, so feel free to use whichever you prefer.